lib/src/client.dart

```diff
@@ -80,6 +97,49 @@ class ClientMethod<Q, R> {
   ClientMethod(this.path, this.requestSerializer, this.responseDeserializer);
 }
 
+/// Runtime options for a RPC call.
+class CallOptions {
+  final Map<String, String> metadata;
+  final Duration timeout;
+
+  CallOptions._(this.metadata, this.timeout);
+
+  factory CallOptions({Map<String, String> metadata, Duration timeout}) {
+    final sanitizedMetadata = <String, String>{};
+    metadata?.forEach((key, value) {
+      final lowerCaseKey = key.toLowerCase();
+      if (!lowerCaseKey.startsWith(':') &&
+          !_reservedHeaders.contains(lowerCaseKey)) {
+        sanitizedMetadata[lowerCaseKey] = value;
+      }
+    });
+    return new CallOptions._(new Map.unmodifiable(sanitizedMetadata), timeout);
+  }
+
+  CallOptions mergedWith(CallOptions other) {
+    if (other == null) return this;
+    final mergedMetadata = new Map.from(metadata)..addAll(other.metadata);
+    final mergedTimeout = other.timeout ?? timeout;
```

Comment:
It's a deliberate choice, yes. The `Client` has a default `CallOptions` that might specify a default timeout, which you can override to something specific per RPC.

In general, anything specified in `other` will override whatever was specified in `this`.